test/schema-special-usage.test.js

```diff
@@ -45,6 +45,53 @@ test('Ajv8 usage instead of the bundle one', t => {
       t.match(err.message, 'strictRequired', 'the new ajv8 option trigger a startup error')
     })
   })
+
+  t.test('use new ajv8 option with JTD schema fails (JTD needs explicit dependency)', t => {
+    t.plan(2)
```

Comment:
@Eomm Is this the kind of test you meant?

Reply:
Yes, but with an example that is expected to work.

Notice that in ajv docs there is a set of limits on the JTD feature like:

> To use discriminator keyword you have to use option discriminator: true with Ajv constructor - it is not enabled by default.

> mapping in discriminator object is not supported.

So, this is a working ajv snippet that we are able to integrate in tests using the `ajv` server option and the `compilersFactory`

```js
const Ajv = require("ajv")

const ajv = new Ajv({ allErrors: true, discriminator: true, strict: false })

const schema = {
  discriminator: { propertyName: "version" },
  oneOf: [
    {
      properties: {
        version: { const: "1" },
        foo: { type: "string" }
      },
      required:['version']
    },
    {
      properties: {
        version: { const: "2" },
        foo: { type: "number" }
      },
      required:['version']
    },
  ],
}

const validate = ajv.compile(schema)

test({ foo: "abc", version: "1" })
test({ foo: 123, version: "1" })
test({ foo: 123, version: "2" })

function test(data) {
  const valid = validate(data)
  if (valid) console.log("Valid!")
  else console.log("Invalid: " + ajv.errorsText(validate.errors))
}
```

https://ajv.js.org/json-schema.html#discriminator

Reply:
@Eomm You're referencing the JSON-Schema documentation. Is the intention to test using JTD https://ajv.js.org/json-type-definition.html#discriminator-form or JSON-Schema?

Reply:
Could you arrange an example?
I get an error on the compile call

Reply:
I'm really not sure what you mean. An example of what exactly? JTD working or failing, or JSON Schema working or failing with JTD set up, or JSON Schema Shorthand working or failing?

Reply:
Your example is just using JSON Schema to test AJVs parsing of JSON Schema discriminators, which has nothing to do with this PR, which only adds a flag to bypass automatic mutation of schema root properties.
I'm really quite confused.

Reply:
Sorry for the late response

> which has nothing to do with this PR

I agree, but I have commented on this test. This PR is adding a test to check a failure that is not related to the `jsonShortHand` feature.

> JTD working or failing, or JSON Schema working or failing with JTD set up, or JSON Schema Shorthand working or failing?

To compose a working test with Fastify, I would start with a plain snippet with `ajv` and a JSON Schema with JTD set up. Then it will be easier to understand how to arrange this specific test